shield.go

```diff
@@ -38,17 +52,17 @@ func digest(password, salt []byte) string {
 // password. This encrypted password is self contained, in the sense
 // that there's no need to store the encrypted password and salt
 // separatedly, as it will have everything in the returned string.
-func Encrypt(password, salt []byte) (string, error) {
-	if len(password) > MaxLength {
+func (s Shield) Encrypt(password, salt []byte) (string, error) {
+	if len(password) > s.maxLength {
 		return "", ErrPasswordTooLong
 	}
-	return digest(password, salt) + string(salt), nil
+	return s.digest(password, salt) + string(salt), nil
 }
 
 // GenerateSalt returns a new salt of SaltLength length filled with
 // random bytes.
-func GenerateSalt() ([]byte, error) {
-	salt := make([]byte, SaltLength)
+func (s Shield) GenerateSalt() ([]byte, error) {
```

Comment:
There's no need to `GenerateSalt` to be a _method_ of `Shield` I think. Let's discuss it.